client/memory.c

```diff
@@ -135,7 +135,7 @@ void *xdag_malloc(size_t size)
 {
 	uint8_t *res;
 	
-	if (size <= 0) {
+	if (size) {
```

Comment:
If size > 0, it will not alloc any memory at all.

Reply:
sorry，that‘s a mistake